ts/src/config.ts

```diff
@@ -77,9 +78,28 @@ export interface Config extends AuthenticationConfig {
   // stack depth may increase overhead of profiling.
   heapMaxStackDepth?: number;
 
-  // Time to wait before trying to create a profile again if profile creation
-  // fails.
-  backoffMillis?: number;
+  // On each consecutive error in profile creation, the backoff envelope will
+  // increase by this factor. The backoff will be a random value selected
+  // from a uniform distribution between 0 and the backoff envelope.
+  backoffMultiplier?: number;
+
+  // On first error during profile creation, if the backoff is not specified
+  // by the server response, then profiler will wait between 0 and
+  // initialBackoffMillis before asking the server to create a profile again.
+  // After a successful profile creation, the backoff envelope will be reset to
+  // initialExpBackoffMillis.
+  initialBackoffMillis?: number;
+
+  // If the backoff is not specified by the server response, then profiler will
+  // wait at most expBackoffMillisCap before asking server to create a profile
+  // again.
+  expBackoffMillisCap?: number;
+
+  // Server-specified backoffs will be capped at serverBackoffMillisCap.
+  // The backoff is capped here because setTimeout (which is used to control
+  // when next profile is collected) will run immediately if the backoff is
+  // to large.
+  serverBackoffMillisCap?: number;
```

Comment:
Unit goes last in the name: serverBackoffCapMillis

Reply:
Done.